Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,486 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+    (e₀ : F.obj zero ⟶ G.obj zero) (e₁ : F.obj one ⟶ G.obj one)
+    {h₁ : F.map left ≫ e₀ = e₁ ≫ G.map left}
+    {h₂ : F.map right ≫ e₀ = e₁ ≫ G.map right}
+    {h₃ : F.map reflexion ≫ e₁ = e₀ ≫ G.map reflexion}
+    {D : Type u₂} [Category.{v₂} D] (H : C ⥤ D) :
+    whiskerRight (mkNatTrans e₀ e₁ : F ⟶ G) H =
+      mkNatTrans (H.map e₀) (H.map e₁)
+          (by simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp, h₁])
+          (by simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp, h₂])
+          (by simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp, h₃]) := by
+  ext x; cases x <;> simp
+
+end
+
+/-- Any functor out of the WalkingReflexivePair yields a reflexive pair -/
+instance to_isReflexivePair {F : WalkingReflexivePair ⥤ C} :
+    IsReflexivePair (F.map .left) (F.map .right) :=
+  ⟨F.map .reflexion, map_reflexion_comp_map_left F, map_reflexion_comp_map_right F⟩
+
+end reflexivePair
+
+/-- A `ReflexiveCofork` is a cocone over a `WalkingReflexivePair`-shaped diagram. -/
+abbrev ReflexiveCofork (F : WalkingReflexivePair ⥤ C) := Cocone F
+
+namespace ReflexiveCofork
+
+open WalkingReflexivePair WalkingReflexivePair.Hom
+
+variable {F : WalkingReflexivePair ⥤ C}
+
+/-- The tail morphism of a reflexive cofork. -/
+abbrev π (G : ReflexiveCofork F) : F.obj zero ⟶ G.pt := G.ι.app zero
+
+/-- Constructor for `ReflexiveCofork` -/
+@[simps pt]
+def mk {X : C} (π : F.obj zero ⟶ X) (h : F.map left ≫ π = F.map right ≫ π) :
+    ReflexiveCofork F where
+  pt := X
+  ι := reflexivePair.mkNatTrans π (F.map left ≫ π)
+
+@[simp]
+lemma mk_π {X : C} (π : F.obj zero ⟶ X) (h : F.map left ≫ π = F.map right ≫ π) :
+    (mk π h).π = π := rfl
+
+lemma condition (G : ReflexiveCofork F) : F.map left ≫ G.π = F.map right ≫ G.π := by
+  erw [Cocone.w G left, Cocone.w G right]
+
+@[simp]
+lemma app_one_eq_π (G : ReflexiveCofork F) : G.ι.app zero = G.π := rfl
+
+/-- The underlying `Cofork` of a `ReflexiveCofork`. -/
+abbrev toCofork (G : ReflexiveCofork F) : Cofork (F.map left) (F.map right) :=
+  Cofork.ofπ G.π (by simp)
+
+end ReflexiveCofork
+
+noncomputable section
+open WalkingReflexivePair WalkingReflexivePair.Hom
+
+variable (F : WalkingReflexivePair ⥤ C)
+
+/-- Forgetting the reflexion yields an equivalence between cocones over a bundled reflexive pair and
+coforks on the underlying parallel pair. -/
+@[simps! functor_obj_pt inverse_obj_pt]
+def reflexiveCoforkEquivCofork :
+    ReflexiveCofork F ≌ Cofork (F.map left) (F.map right) :=
+  (Functor.Final.coconesEquiv _ F).symm.trans (Cocones.precomposeEquivalence
+    (diagramIsoParallelPair (WalkingParallelPair.inclusionWalkingReflexivePair ⋙ F))).symm
+
+@[simp]
+lemma reflexiveCoforkEquivCofork_functor_obj_π (G : ReflexiveCofork F) :
+    ((reflexiveCoforkEquivCofork F).functor.obj G).π = G.π := by
+  dsimp [reflexiveCoforkEquivCofork]
+  rw [ReflexiveCofork.π, Cofork.π]
+  aesop_cat
+
+@[simp]
+lemma inclusionWalkingReflexivePairEquivCocone_inverse_obj_π
```

Comment:
```suggestion
lemma reflexiveCoforkEquivCofork_inverse_obj_π
```
I will try to see if there is a better proof...

Reply:
Could you add the following lemma in `Functor.Final`:
```lean
/-- Alternative equational lemma for `(extendCocone c).ι.app` in case a lift of the object
is given explicitly. -/
lemma extendCocone_obj_ι_app' (c : Cocone (F ⋙ G)) {X : D} {Y : C} (f : X ⟶ F.obj Y) :
    (extendCocone.obj c).ι.app X = G.map f ≫ c.ι.app Y := by
  sorry
```
and then the proof here can be:
```lean
@[simp]
lemma reflexiveCoforkEquivCofork_inverse_obj_π
    (G : Cofork (F.map left) (F.map right)) :
    ((reflexiveCoforkEquivCofork F).inverse.obj G).π = G.π := by
  dsimp only [reflexiveCoforkEquivCofork, Equivalence.symm, Equivalence.trans,
    ReflexiveCofork.π, Cocones.precomposeEquivalence, Cocones.precompose,
    Functor.comp, Functor.Final.coconesEquiv]
  rw [Functor.Final.extendCocone_obj_ι_app' (Y := .one) (f := 𝟙 zero)]
  simp
```